pkg/clusters/addons/kong/kong.go

```diff
@@ -296,3 +360,75 @@ func urlForService(ctx context.Context, cluster clusters.Cluster, nsn types.Name
 
 	return nil, fmt.Errorf("service %s has not yet been provisoned", service.Name)
 }
+
+func deployKongEnterpriseLicenseSecret(ctx context.Context, cluster clusters.Cluster, namespace, name string) error {
+	license := os.Getenv("KONG_ENTERPRISE_LICENSE")
+	if license == "" {
+		return fmt.Errorf("failed kong enterprise license from environment setting")
+	}
+
+	newSecret := &corev1.Secret{
+		Type: corev1.SecretTypeOpaque,
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			"license": []byte(license),
+		},
+	}
+
+	_, err := cluster.Client().CoreV1().Secrets(namespace).Create(ctx, newSecret, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed creating kong-enterprise-license secret, err %w", err)
+	}
+
+	fmt.Printf("successfully deployed kong-enterprise-license into the cluster .")
+	return nil
+}
+
+func prepareSecrets(ctx context.Context, namespace string) error {
+	stderr := new(bytes.Buffer)
+	pwd := fmt.Sprintf("--from-literal=password=%s", EnterpriseKongAdminDefaultPWD)
```

Comment:
Question: At this level, we seem to be losing track of the `a.kongAdminPassword` attribute, but I see that we appear to be passing this in as `fmt.Sprintf("env.kong_password=%s", a.kongAdminPassword)` above. I'm actually not sure if this works, does this conflict? It seems like if you set the secret, you don't need to set the `env` right?

Reply:
It is using the password as a seed to the secret, hope this is clear.

Reply:
based on this doc https://docs.konghq.com/enterprise/2.5.x/admin-api/rbac/examples/